docs/content/dev/2.0.md

```diff
@@ -14,6 +14,12 @@ The software provides two special groups which have no explicit users:
 - `everyone` (Describing that everyone who wants to access a note can do if it is enabled in the config.)
 - `loggedIn` (Describing all users which are logged in)
 
+## Deleting notes and revisions
+- The owner of a note may delete it. This also removes all revisions and all files
+  that were uploaded to that note.
```

Comment:
I'm not sure how ideal this will work for some use-cases. Images that are uploaded to a document might be re-used in other documents, since it's normal in Markdown to do that. This would only make sense if we inline pictures into markdown.

Otherwise I would consider to opt for attachments being managed independently from note deletions.

Reply:
A thing that just occurred to me:
What about uploads of other people to my note, that I want to delete? Should they be removed, too? Or should they just lose their note attachment? I mean on the one hand they uploaded an image to somebody else's note and therefore one could argue if the note get's deleted so should the image, but I'm not sure how well that flies.

Reply:
IIRC, the idea we contemplated to solve that was to let the frontend automatically re-upload images that are copy-pasted between notes, but granted, that might be a somewhat fragile solution.

I really like to have uploads attached to notes, as that provides a simple way to delete all data associated with a note. I'd argue that this behavior is more in line with what a user expects when pressing the delete button. The frontend could make this more obvious by showing a list of uploads that will be deleted when deleting the note.

Reply:
We could also make deleting the associated uploads optional. I'm not super happy with that solution, as a user might forget they didn't delete an image which then stays online forever, but I guess that can be mitigated by the "files uploaded by me" list in the user profile.

Reply:
From frontend perspective the first thing that comes to my mind would be an additional checkbox in the note deletion modal asking whether attached images should be removed too.
This way a user that wants to delete their note has to think about attached images.

Reply:
As a User I would definitely not expect to have images deleted when I delete a note. It's not the first time, that I would copy a note and continue to use that content independent from the original note. Means my image links would still be the same. Unless you want to write a parser that re-write the image URL inside my note, I don't think this would work.

And if you do, I would be quite unhappy that the application I use to write markdown, simply decides what image URLs I should use.

Reply:
But the current situation of once uploaded you basically can't remove images ever is also not feasible. From a legal perspective at least. I think giving the option to delete media that's associated with a note is an okay way to go.

Reply:
I would rather associate media with users who uploaded it. This would leave the user in control, without breaking notes unexpectedly.

I mean, we need this anyway, when we decide not to require the media deletion for notes.

Reply:
Currently an upload is associated with a user and a note.